extensions-core/namespace-lookup/src/main/java/io/druid/server/lookup/namespace/cache/OffHeapNamespaceExtractionCacheManager.java

```diff
@@ -83,27 +76,32 @@ public OffHeapNamespaceExtractionCacheManager(
         .commitFileSyncDisable()
         .cacheSize(10_000_000)
         .make();
-    lifecycle.addHandler(
-        new Lifecycle.Handler()
-        {
-          @Override
-          public void start() throws Exception
+    try {
+      lifecycle.addMaybeStartHandler(
+          new Lifecycle.Handler()
           {
-            // NOOP
-          }
+            @Override
+            public void start() throws Exception
+            {
+              // NOOP
+            }
 
-          @Override
-          public void stop()
-          {
-            if (!mmapDB.isClosed()) {
-              mmapDB.close();
-              if (!tmpFile.delete()) {
-                log.warn("Unable to delete file at [%s]", tmpFile.getAbsolutePath());
+            @Override
+            public void stop()
+            {
+              if (!mmapDB.isClosed()) {
```

Comment:
wondering, this doesn't need to be thread safe ? are we sure that only one thread will call stop ?